str.go

```diff
@@ -329,25 +329,37 @@ func StrSliceToRRuleSet(ss []string) (*Set, error) {
 	return &set, nil
 }
 
-// StrToDates accepts string with format: "VALUE=DATE-TIME:{time},{time},...,{time}"
+// StrToDates is inteded to parse RDATE and EXDATE properties supporting only
+// VALUE=DATE-TIME (DATE and PERIOD are not supported).
+// Accepts string with format: "VALUE=DATE-TIME;[TZID=...]:{time},{time},...,{time}"
 // or simply "{time},{time},...{time}" and parses it to array of dates
-// may be used to parse RDATE/EXDATE rules
 func StrToDates(str string) (ts []time.Time, err error) {
 	tmp := strings.Split(str, ":")
 	if len(tmp) > 2 {
 		return nil, fmt.Errorf("bad format")
 	}
+	var loc *time.Location
 	if len(tmp) == 2 {
 		params := strings.Split(tmp[0], ";")
 		for _, param := range params {
-			if param != "VALUE=DATE-TIME" {
-				return nil, fmt.Errorf("unsupported RDATE/EXDATE parm: %v", param)
+			if strings.HasPrefix(param, "TZID=") {
+				loc, err = parseTZID(param)
+			} else if param != "VALUE=DATE-TIME" {
+				err = fmt.Errorf("unsupported RDATE/EXDATE parm: %v", param)
+			}
+			if err != nil {
+				return
```

Comment:
The other error-case returns in this function do not use naked returns.

Reply:
OK. Fixed.